webapp/graphite/tags/redis.py

```diff
@@ -0,0 +1,167 @@
+from __future__ import absolute_import
+
+import re
+
+from django.conf import settings
+
+from graphite.tags.utils import BaseTagDB, TaggedSeries
+
+class RedisTagDB(BaseTagDB):
+  """
+  Stores tag information in a Redis database.
+
+  Keys used are:
+
+  .. code-block:: none
+
+    series                    # Set of all paths
+    series:<path>:tags        # Hash of all tag:value pairs for path
+    tags                      # Set of all tags
+    tags:<tag>:series         # Set of paths with entry for tag
+    tags:<tag>:values         # Set of values for tag
+    tags:<tag>:values:<value> # Set of paths matching tag/value
+
+  """
+  def __init__(self):
+    from redis import Redis
+
+    self.r = Redis(host=settings.TAGDB_REDIS_HOST,port=settings.TAGDB_REDIS_PORT,db=settings.TAGDB_REDIS_DB)
+
+  def find_series(self, tags):
+    with self.r.pipeline() as pipe:
+      all_match_empty = True
+
+      for tagspec in tags:
+        m = re.match('^([^;!=]+)(!?=~?)([^;]*)$', tagspec)
```

Comment:
This regex is defined twice, in `webapp/graphite/tags/localdatabase.py` as well. Might make sense to move them into one definition, just to make sure that future changes get applied to every location where they are needed.